### PR TITLE
Fixes estimated Pending Rewards value decreased unexpectedly - follow up to #16678 - 1.27.x

### PR DESCRIPTION
**Pull request by tmancey**
Uplift of https://github.com/brave/brave-core/pull/9418
Uplift of https://github.com/brave/brave-core/pull/9477
Resolves https://github.com/brave/brave-browser/issues/16861

- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR.
- [x] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Review by fmarier — APPROVED** [review]
I didn't look at this too closely, but approving anyways since I'm still listed as a codeowner of some of these files on the 1.27.x branch and don't want to block this.

**Comment by kjozwiak** [comment]
`post-init` failed due to several `pylint` as per https://ci.brave.com/job/pr-brave-browser-issues-16861-1.27.x-post-init/4/pylint/. However, we usually don't block uplifts due to `pylint` errors/warnings.

Restarting `Android` due to `dist` failing as per https://ci.brave.com/job/pr-brave-browser-issues-16861-1.27.x-android/4/execution/node/161/log/.

```
 Do you agree to our data usage policy? (y/n) -->Traceback (most recent call last):
17:04:22    File "/home/ubuntu/workspace/pr-brave-browser-issues-16861-1.27.x-android/src/brave/vendor/depot_tools/.cipd_bin/goma_auth.py", line 573, in <module>
17:04:22      sys.exit(main())
17:04:22    File "/home/ubuntu/workspace/pr-brave-browser-issues-16861-1.27.x-android/src/brave/vendor/depot_tools/.cipd_bin/goma_auth.py", line 569, in main
17:04:22      return action()
17:04:22    File "/home/ubuntu/workspace/pr-brave-browser-issues-16861-1.27.x-android/src/brave/vendor/depot_tools/.cipd_bin/goma_auth.py", line 411, in Login
17:04:22      ConfirmUserAgreedToS()
17:04:22    File "/home/ubuntu/workspace/pr-brave-browser-issues-16861-1.27.x-android/src/brave/vendor/depot_tools/.cipd_bin/goma_auth.py", line 95, in ConfirmUserAgreedToS
17:04:22      yn = INPUT('Do you agree to our data usage policy? (y/n) -->')
17:04:22 
 EOFError: EOF when reading a line
```

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.27.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/9418#issuecomment-881707990.

**Pull request merged**
